rust/agama-server/src/cert.rs

```diff
@@ -24,15 +27,35 @@ use openssl::x509::{X509NameBuilder, X509};
 //     pub write(...)
 // }
 
+pub const DEFAULT_CERT_DIR: &str = "/etc/agama.d/ssl";
+
+/// Writes the certificate and the key to the well known location
+pub fn write_certificate(cert: X509, key: PKey<Private>) -> anyhow::Result<()> {
+    // check and create default dir if needed
+    if ! Path::new(DEFAULT_CERT_DIR).is_dir() {
+        std::fs::create_dir_all(DEFAULT_CERT_DIR)?;
+    }
+
+    if let Ok(bytes) = cert.to_pem() {
+        fs::write(Path::new(DEFAULT_CERT_DIR).join("cert.pem"), bytes)?;
+    }
+    if let Ok(bytes) = key.public_key_to_pem() {
```

Comment:
So , `key` is supposed to be the _private_ key, yet we write it with a "_public_ key to PEM" method??

Reply:
already fixed in local branch, but thanks for pointing that ;-)

Reply:
Okay... why don't you push what you have then? No one likes reviewing outdated code :smiling_face_with_tear:

Reply:
done